src/components/Nav/Nav.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from 'react';
+import './Nav.scss';
+import { Link } from 'react-router-dom';
+import './NavSmall';
+import NavSmall from './NavSmall';
+
+// true false
+const Navv = () => {
+  const [show, setShow] = useState(true);
+  const [middleOn, setMiddleOn] = useState(false);
+
+  useEffect(() => {
+    window.addEventListener('scroll', () => {
+      window.scrollY > 1 ? setShow(false) : setShow(true);
+    });
+    return () => {
+      window.removeEventListener('scroll');
+    };
+  }, []);
+
+  const handleSearchOpen = () => {
+    middleOn ? setMiddleOn(false) : setMiddleOn(true);
+  };
```

Comment:
```suggestion
  const handleSearchOpen = () => {
    setMiddleOn(prev => !prev);
  };
```
해당 함수는 토글 형태로 결국 search를 누르면 검색창이 나타나고
x 버튼을 누르면 닫는 형태기 때문에 이와같이 구현해줄 수 있겠습니다.